src/Exception/InvalidColumnDeclaration.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Exception;
+
+use Doctrine\DBAL\Exception;
+
+use function sprintf;
+
+/** @psalm-immutable */
+final class InvalidColumnDeclaration extends Exception
+{
+    public static function fromInvalidColumnType(string $columnName, InvalidColumnType $e): self
+    {
+        return new self(sprintf(
+            'Column "%s" has invalid type: %s',
```

Comment:
I wouldn't repeat `$e->getMessage()` here, it's redundant. In my experience, it leads to people examining the wrong stack trace because they think the intermediary exception is somehow better because it has a more complete message.